src/runtime/virtcontainers/fs_share_linux.go

```diff
@@ -455,13 +457,94 @@ func (f *FilesystemShare) shareRootFilesystemWithNydus(ctx context.Context, c *C
 	f.Logger().Infof("Nydus rootfs info: %#v\n", rootfs)
 
 	return &SharedFile{
-		storage:   rootfs,
-		guestPath: rootfsGuestPath,
+		containerStorages: []*grpc.Storage{rootfs},
+		guestPath:         rootfsGuestPath,
+	}, nil
+}
+
+// handleVirtualVolume processes each `extraoption` in rootFs.Options,
+// creating storage, and then aggregates all storages  into an array.
+func handleVirtualVolume(c *Container) ([]*grpc.Storage, string, error) {
+	var volumes []*grpc.Storage
+	var volumeType string
+
+	for _, o := range c.rootFs.Options {
+		if strings.HasPrefix(o, VirtualVolumePrefix) {
+			virtVolume, err := types.ParseKataVirtualVolume(strings.TrimPrefix(o, VirtualVolumePrefix))
+			if err != nil {
+				return nil, "", err
+			}
+
+			volumeType = virtVolume.VolumeType
+			var vol *grpc.Storage
+			///TODO implement the logic with KataVirtualVolume types
+			switch volumeType {
+			case types.KataVirtualVolumeImageGuestPullType:
+				vol, err = handleVirtualVolumeStorageObject(c, "", virtVolume)
+				if err != nil {
+					return nil, "", err
+				}
+			case types.KataVirtualVolumeImageRawBlockType, types.KataVirtualVolumeLayerRawBlockType:
+				vol, err = handleVirtualVolumeStorageObject(c, "", virtVolume)
+				if err != nil {
+					return nil, "", err
+				}
+			case types.KataVirtualVolumeImageNydusBlockType, types.KataVirtualVolumeLayerNydusBlockType:
+				vol, err = handleVirtualVolumeStorageObject(c, "", virtVolume)
+				if err != nil {
+					return nil, "", err
+				}
+			case types.KataVirtualVolumeImageNydusFsType, types.KataVirtualVolumeLayerNydusFsType:
+				vol, err = handleVirtualVolumeStorageObject(c, "", virtVolume)
+				if err != nil {
+					return nil, "", err
+				}
+			case types.KataVirtualVolumeDirectBlockType:
+				vol, err = handleVirtualVolumeStorageObject(c, "", virtVolume)
+				if err != nil {
+					return nil, "", err
+				}
+			default:
+			}
+			if vol != nil {
+				volumes = append(volumes, vol)
+			}
+		}
+	}
+
+	return volumes, volumeType, nil
+}
+
+func (f *FilesystemShare) shareRootFilesystemWithVirtualVolume(ctx context.Context, c *Container) (*SharedFile, error) {
+	guestPath := filepath.Join("/run/kata-containers/", c.id, c.rootfsSuffix)
+	rootFsStorages, volumeType, err := handleVirtualVolume(c)
+	if err != nil {
+		return nil, err
+	}
+
+	///TODO implement the logic with KataVirtualVolume types
+	switch volumeType {
+	case types.KataVirtualVolumeImageGuestPullType:
+	case types.KataVirtualVolumeImageRawBlockType, types.KataVirtualVolumeLayerRawBlockType:
+	case types.KataVirtualVolumeImageNydusBlockType, types.KataVirtualVolumeLayerNydusBlockType:
+	case types.KataVirtualVolumeImageNydusFsType, types.KataVirtualVolumeLayerNydusFsType:
+	case types.KataVirtualVolumeDirectBlockType:
+	default:
+	}
```

Comment:
Hey Chengyu, you have a couple of these volumeType switch blocks with TODOs, which I'm wondering about. 
They don't seem to be doing anything at the moment, so are they set-up for a future PR? Would it be worth saving these blocks to that PR rather than adding code that doesn't do anything right now? Thanks!

Reply:
>They don't seem to be doing anything at the moment, so are they set-up for a future PR? 

Hey Steve - I appreciate your review and feedback. These code are set-up for PRs(https://github.com/kata-containers/kata-containers/pull/8484, https://github.com/kata-containers/kata-containers/pull/7837). I would fix it right now.

Reply:
Fixed.